Automattic/wp-calypso
# useGetDialogText: Refactor to account for single-countable limitation

This pull request updates the implementation of the `useGetDialogText` hook to compensate for a limitation in the way translations are processed. It's currently not possible to translate strings with more than one countable value, such as when handling multiple plugins affecting multiple sites. To work around this, we split the translation into two strings and concatenate them.

Resolves `1202619025189113-as-1205234897865177`.
Related to #79135.

## Proposed Changes

* Re-work the internal types/interfaces used for retrieving and translating text for the plugin action modal.
* Organize translations into separate files, grouped by action (e.g., activation, installation, etc.)

## Testing Instructions

The testing procedure for this PR is identical to #79135. Instructions are copied below for convenience:

---

***TL;DR:** Regression testing for all plugin management functions.*

* In the Jetpack Cloud dashboard for agencies/partners, experiment with the various available plugin management actions:
  * Try affecting just one plugin, just one site, multiples of either, or multiples of both.
  * Try accepting the confirmation modal, and try canceling it.
  * Pay special attention to the modal's verbiage to check that the action name, site title and/or plugin name (if applicable), and spelling are all correct.
* In Calypso Blue, do much the same testing on the `/plugins/{pluginSlug}` and `/plugins/{pluginSlug}/{siteSlug}` pages. (NOTE: You'll likely need access to a WPCOM Business or eCommerce site for this step.) Confirm that all functionality works as expected.

### Reference screenshots

<img width="597" alt="image" src="https://github.com/Automattic/wp-calypso/assets/670067/d3cdacb9-24a5-4e9d-b4b4-9faf7f21a1ca">

<img width="918" alt="image" src="https://github.com/Automattic/wp-calypso/assets/670067/40484051-ee7a-4615-8db4-e5b04b1256cc">

<img width="1202" alt="image" src="https://github.com/Automattic/wp-calypso/assets/670067/7d6f9da7-942e-4ad4-acf1-120579a43ee0">

<img width="1318" alt="image" src="https://github.com/Automattic/wp-calypso/assets/670067/69f9bf94-36fa-4fe3-a386-810267c6119f">
